core/src/main/scala/kafka/utils/DelayedItem.scala

```diff
@@ -21,24 +21,24 @@ import java.util.concurrent._
 
 import org.apache.kafka.common.utils.Time
 
-import scala.math._
+class DelayedItem(val delayMs: Long, val time: Time) extends Logging {
+  def this(delayMs: Long) = this(delayMs, Time.SYSTEM)
 
-class DelayedItem(val delayMs: Long) extends Delayed with Logging {
-
-  private val dueMs = Time.SYSTEM.milliseconds + delayMs
-
-  def this(delay: Long, unit: TimeUnit) = this(unit.toMillis(delay))
+  private val dueNs = time.nanoseconds + TimeUnit.MILLISECONDS.toNanos(delayMs)
```

Comment:
Should we handle the overflow?

Reply:
Any idea how to efficiently handle the overflow here? I think it's very unlikely we'll ever hit it given our normal delays.

Reply:
> I think it's very unlikely we'll ever hit it given our normal delays.

you are right :)